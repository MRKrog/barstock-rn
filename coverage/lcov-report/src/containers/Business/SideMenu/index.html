<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/containers/Business/SideMenu</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> src/containers/Business/SideMenu
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">35% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">22.22% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/18</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="SideMenu.js"><a href="SideMenu.js.html">SideMenu.js</a></td>
	<td data-value="35" class="pic low"><div class="chart"><div class="cover-fill" style="width: 35%;"></div><div class="cover-empty" style="width:65%;"></div></div></td>
	<td data-value="35" class="pct low">35%</td>
	<td data-value="20" class="abs low">7/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="9" class="abs low">2/9</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="18" class="abs low">6/18</td>
	</tr>

<tr>
	<td class="file empty" data-value="SideMenu.style.js"><a href="SideMenu.style.js.html">SideMenu.style.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed May 29 2019 20:10:29 GMT-0600 (Mountain Daylight Time)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
<script src="../../../../block-navigation.js"></script>
</body>
</html>
