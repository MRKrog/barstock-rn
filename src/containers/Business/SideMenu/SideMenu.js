import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './SideMenu.style';
import call from 'react-native-phone-call';
import email from 'react-native-email';
import { NavigationActions } from 'react-navigation';
import { ScrollView, Text, View, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

class SideMenu extends Component {
  navigateToScreen = (route) => () => {
    const navigateAction = NavigationActions.navigate({
      routeName: route
    });
    let { routeName } = this.props.navigation.state;
    console.log(this.props.navigation.state.routes);

    this.props.navigation.dispatch(navigateAction);
  }

  makeCall = (number) => {
   const args = {
     number: "6307154311",
     prompt: true
   }
    call(args).catch(console.error)
  }

  handleEmail = () => {
    const to = ['tiaan@email.com']
    email(to, {
      cc: ['bazzy@moo.com', 'doooo@daaa.com'],
      subject: 'Bar Email',
      body: 'Body'
    }).catch(console.error)
  }


  render () {
    return (
      <View style={styles.container}>
        <View>
          <Text style={styles.sectionHeadingStyle}></Text>
          <View style={styles.navSectionStyle}>
            <TouchableOpacity style={styles.navItemContainer} onPress={this.navigateToScreen('Account')}>
              <Icon name='user-circle' color='#ffffff' size={20} style={styles.iconStyle} />
              <Text style={styles.navItemStyle}>Account Info</Text>
            </TouchableOpacity>

            <TouchableOpacity style={styles.navItemContainer} onPress={this.navigateToScreen('Order')}>
              <Icon name='shopping-cart' color='#ffffff' size={20} style={styles.iconStyle} />
              <Text style={styles.navItemStyle}>Inventory</Text>
            </TouchableOpacity>

            <TouchableOpacity style={styles.navItemContainer} onPress={this.navigateToScreen('BarInventory')}>
              <Icon name='edit' color='#ffffff' size={20} style={styles.iconStyle} />
              <Text style={styles.navItemStyle}>Bar Items</Text>
            </TouchableOpacity>

            <TouchableOpacity style={styles.navItemContainer} onPress={this.navigateToScreen('PastOrders')}>
              <Icon name='history' color='#ffffff' size={20} style={styles.iconStyle} />
              <Text style={styles.navItemStyle}>Past Orders</Text>
            </TouchableOpacity>

            <TouchableOpacity style={styles.navItemContainer} onPress={this.navigateToScreen('Login')}>
              <Icon name='sign-out' color='#ffffff' size={20} style={styles.iconStyle} />
              <Text style={styles.navItemStyle}>Logout</Text>
            </TouchableOpacity>
          </View>
        </View>

        <View style={styles.footerContainer}>
          <TouchableOpacity onPress={()=> this.makeCall()}>
            <Icon name="phone" size={30} color="#332D2F" />
          </TouchableOpacity>
          <TouchableOpacity onPress={()=> this.handleEmail()}>
            <Icon name="envelope-o" size={30} color="#332D2F" />
          </TouchableOpacity>
        </View>

      </View>
    );
  }
}

SideMenu.propTypes = {
  navigation: PropTypes.object
};

export default SideMenu;
