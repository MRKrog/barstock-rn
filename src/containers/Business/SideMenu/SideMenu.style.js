export default {
  container: {
    display: "flex",
    justifyContent: "space-between",
    paddingTop: 20,
    flex: 1,
    backgroundColor: '#231f20'
  },
  navItemContainer: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "flex-start",
    alignItems: "center",
    paddingHorizontal: 20,
    paddingVertical: 20,
    // borderBottomColor: '#ececec',
    // borderBottomWidth: .2,
  },
  iconStyle: {
    marginRight: 15
  },
  navItemStyle: {
    fontSize: 18,
    textTransform: "uppercase",
    fontFamily: 'abel',
    textAlign: "left",
    color: "#ffffff",
    letterSpacing: 1,
  },
  navSectionStyle: {

  },
  sectionHeadingStyle: {
    paddingVertical: 10,
    paddingHorizontal: 5,
  },
  footerContainer: {
    paddingVertical: 20,
    backgroundColor: "#161515",
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-evenly",
    shadowOpacity: 0.3,
    shadowRadius: 3,
    shadowColor: '#161515',
    shadowOffset: { height: -3, width: 0 },
    zIndex: 9,
  }
};
