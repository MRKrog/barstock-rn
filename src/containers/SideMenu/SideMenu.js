import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './SideMenu.style';
import { NavigationActions } from 'react-navigation';
import { ScrollView, Text, View } from 'react-native';

class SideMenu extends Component {
  navigateToScreen = (route) => () => {
    const navigateAction = NavigationActions.navigate({
      routeName: route
    });
    this.props.navigation.dispatch(navigateAction);
  }

  render () {
    return (
      <View style={styles.container}>
        <ScrollView>
          <View>
            <Text style={styles.sectionHeadingStyle}></Text>
            <View style={styles.navSectionStyle}>
              <Text style={styles.navItemStyle} onPress={this.navigateToScreen('Home')}>
              Home
              </Text>
              <Text style={styles.navItemStyle} onPress={this.navigateToScreen('Login')}>
              Logout
              </Text>
            </View>
          </View>
        </ScrollView>
        <View style={styles.footerContainer}>
          <Text></Text>
        </View>
      </View>
    );
  }
}

SideMenu.propTypes = {
  navigation: PropTypes.object
};

export default SideMenu;
