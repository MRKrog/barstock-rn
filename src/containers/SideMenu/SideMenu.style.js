export default {
  container: {
    paddingTop: 20,
    flex: 1,
    backgroundColor: '#231f20'
  },
  navItemStyle: {
    fontSize: 20,
    textTransform: "uppercase",
    fontFamily: 'abel',
    textAlign: "left",
    color: "#ffffff",
    letterSpacing: 1,
    padding: 15
  },
  navSectionStyle: {

  },
  sectionHeadingStyle: {
    paddingVertical: 10,
    paddingHorizontal: 5,
  },
  footerContainer: {
    padding: 20,
    backgroundColor: "#ffffff"
  }
};
