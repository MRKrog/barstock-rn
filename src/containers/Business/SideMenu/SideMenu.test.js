import { shallow } from "enzyme"
import { SideMenu } from "./SideMenu"
import React from "react"

describe("SideMenu", () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <SideMenu />
    )
  })

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
