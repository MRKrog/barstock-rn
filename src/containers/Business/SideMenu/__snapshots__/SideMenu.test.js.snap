// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#231f20",
      "display": "flex",
      "flex": 1,
      "justifyContent": "space-between",
      "paddingTop": 20,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "paddingHorizontal": 5,
          "paddingVertical": 10,
        }
      }
    />
    <View
      style={Object {}}
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="#ffffff"
          name="user-circle"
          size={20}
          style={
            Object {
              "marginRight": 15,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "abel",
              "fontSize": 18,
              "letterSpacing": 1,
              "textAlign": "left",
              "textTransform": "uppercase",
            }
          }
        >
          Account Info
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="#ffffff"
          name="shopping-cart"
          size={20}
          style={
            Object {
              "marginRight": 15,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "abel",
              "fontSize": 18,
              "letterSpacing": 1,
              "textAlign": "left",
              "textTransform": "uppercase",
            }
          }
        >
          Inventory
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="#ffffff"
          name="edit"
          size={20}
          style={
            Object {
              "marginRight": 15,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "abel",
              "fontSize": 18,
              "letterSpacing": 1,
              "textAlign": "left",
              "textTransform": "uppercase",
            }
          }
        >
          Bar Items
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="#ffffff"
          name="history"
          size={20}
          style={
            Object {
              "marginRight": 15,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "abel",
              "fontSize": 18,
              "letterSpacing": 1,
              "textAlign": "left",
              "textTransform": "uppercase",
            }
          }
        >
          Past Orders
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          color="#ffffff"
          name="sign-out"
          size={20}
          style={
            Object {
              "marginRight": 15,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#ffffff",
              "fontFamily": "abel",
              "fontSize": 18,
              "letterSpacing": 1,
              "textAlign": "left",
              "textTransform": "uppercase",
            }
          }
        >
          Logout
        </Text>
      </TouchableOpacity>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#161515",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "paddingVertical": 20,
        "shadowColor": "#161515",
        "shadowOffset": Object {
          "height": -3,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
        "zIndex": 9,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Icon
        allowFontScaling={false}
        color="#332D2F"
        name="phone"
        size={30}
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Icon
        allowFontScaling={false}
        color="#332D2F"
        name="envelope-o"
        size={30}
      />
    </TouchableOpacity>
  </View>
</View>
`;
